### ISSUE TITLE
Should AbstractNode marked as abstract class/interface

### ISSUE THREAD

**Issue by bitPogo** [asciidoctor/asciidoctor.js#1117]
According to the dec block in the types as well as in the code AbstractNode is a abstract BaseClass. For more semantical expression it should be marked as a abstract class in the type definition.

**Comment by Mogztter** [comment]
Indeed, even though you cannot really extend `AbstractNode` or at least that's not the right way to do it since `AbstractNode` is a Ruby class. So if you want to extend `AbstractNode` you will need to create a new class with `AbstractNode` as a superclass using `Opal.allocate_class` (from Opal runtime):

https://github.com/opal/opal/blob/505fd5d7e899a6766472b302c079b4d383fd8216/opal/corelib/runtime.js#L452

**Comment by bitPogo** [comment]
Thats true! Maybe this could be compensated by adding [`@sealed`](https://netbasal.com/behind-the-scenes-how-typescript-decorators-operate-28f8dcacb224).
Also this applies for most the classes (maybe even for all of them) in this project.
